nums/numpy/api.py

```diff
@@ -398,6 +410,11 @@ def allclose(a: BlockArray, b: BlockArray, rtol=1.e-5, atol=1.e-8, equal_nan=Fal
     return _instance().allclose(a, b, rtol, atol)
 
 
+def array_equal(a1: BlockArray, a2: BlockArray, equal_nan=False) -> BlockArray:
```

Comment:
I think this implementation is okay, but please test the semantics of `array_equal` rigorously to ensure that `allclose` with 0 tolerance is an equivalent function to `array_equal`. Alternatively, we can discuss how to generalize the `allclose` function in application to support both `allclose` and `array_equal`.

Reply:
`allclose` uses the following equation: `absolute(a - b) <= (atol + rtol * absolute(b))`. It's typically used for float arrays to prevent floating point error, while `array_equal` is meant for integer arrays. Keeping those usage cases in mind, I believe this implementation should be okay, right? I have a test in `test_array_equal` that has a very small difference in floating points, and `array_equal` returns false as expected.

Reply:
Okay. To be safe, please test edge cases such as positive/negative zero, behavior for pos/neg infinity, and comparisons between every combination of bool, int, float, and complex types (e.g. is `False` equal to `0.0`).

Reply:
Will do.